week2/homework/exercises/1-blog-API/server.js

```diff
@@ -1,10 +1,94 @@
-const express = require('express')
+const express = require("express");
 const app = express();
- 
 
 // YOUR CODE GOES IN HERE
-app.get('/', function (req, res) {
-  res.send('Hello World')
-})
- 
-app.listen(3000)
\ No newline at end of file
+const fs = require("fs");
+const path = require("path");
+const bodyParser = require("body-parser");
+const { title } = require("process");
+
+app.use(express.json());
+
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(bodyParser.json());
+
+app.use("/blogs", express.static(path.join(__dirname, "blogs")));
+
+// app.get("/", function (req, res) {
+//   res.send("Ismaiel says: 'Hello World'");
+// });
+// function
+function isDefined(word) {
+  if (word === undefined) {
+    return false;
+  } else {
+    return true;
+  }
+}
+
+// create a blog
+app.post("/", (req, res) => {
+  let title = req.body.title;
+  let content = req.body.content;
+  if (isDefined(title) || isDefined(content)) {
+    fs.writeFileSync(path.join(__dirname, "./blogs", title), content);
+    res.end(`file ${title} has created`);
+  } else {
+    console.log("Please enter correct data");
```

Comment:
the users can not see the server console
```suggestion
    res.status(400).send('Title and content required');
```

Reply:
Yes, you are right,
I have changed it to res.send().
Thanks for this comment.